libexec/python/docker/api.py

```diff
@@ -295,12 +297,22 @@ def get_layer(image_id,namespace,repo_name,download_folder=None,registry=None,au
         # Update user what we are doing
         print("Downloading layer %s" %image_id)
 
-    response = api_get(base,headers=token,stream=download_folder)
-    if isinstance(response, HTTPError):
-        logger.error("Error downloading layer %s, exiting.", base)
-        sys.exit(1)
-
-    return response
+    try:
+        # Create temporary file with format .tar.gz.tmp.XXXXX
+        fd, tmp_file = tempfile.mkstemp(prefix=("%s.tmp." % download_folder))
+        os.close(fd)
+        response = api_get(base,headers=token,stream=tmp_file)
+        if isinstance(response, HTTPError):
+            sys.exit(1)
+        os.rename(tmp_file, download_folder)
+    except BaseException as e:
```

Comment:
This looks great! I just tested walking through a layer download manually, and then from command line with a clean install. Two quick edits before merge - could we have a sys.exit(1) here (to replace the `BaseException`) as well, and then for each of the exits, since we lose the information that a specific error would provide, could we have the logger print an error so the user can see the specific error with `--debug`. So for the first maybe something like:

      if isinstance(response,HTTPError):
          logger.error("Error downloading image layer %s, exiting.", base)
          sys.exit(1)

Then for the second, maybe:

      except:
            logger.error("Error creating temporary file %s in cache, exiting.",temp_file)
            sys.exit(1)

Feel free to tweak the language to what you think is most informative.